src/ai/models.py

```diff
@@ -0,0 +1,187 @@
+from torchvision.transforms import InterpolationMode, transforms
+import torch
+
+
+txt_animalclasses = {
+    "fr": [
+        "blaireau",
+        "bouquetin",
+        "cerf",
+        "chamois",
+        "chat",
+        "chevre",
+        "chevreuil",
+        "chien",
+        "ecureuil",
+        "equide",
+        "genette",
+        "herisson",
+        "lagomorphe",
+        "loup",
+        "lynx",
+        "marmotte",
+        "micromammifere",
+        "mouflon",
+        "mouton",
+        "mustelide",
+        "oiseau",
+        "ours",
+        "ragondin",
+        "renard",
+        "sanglier",
+        "vache",
+    ],
+    "en": [
+        "badger",
+        "ibex",
+        "red deer",
+        "chamois",
+        "cat",
+        "goat",
+        "roe deer",
+        "dog",
+        "squirrel",
+        "equid",
+        "genet",
+        "hedgehog",
+        "lagomorph",
+        "wolf",
+        "lynx",
+        "marmot",
+        "micromammal",
+        "mouflon",
+        "sheep",
+        "mustelid",
+        "bird",
+        "bear",
+        "nutria",
+        "fox",
+        "wild boar",
+        "cow",
+    ],
+    "it": [
+        "tasso",
+        "stambecco",
+        "cervo",
+        "camoscio",
+        "gatto",
+        "capra",
+        "capriolo",
+        "cane",
+        "scoiattolo",
+        "equide",
+        "genet",
+        "riccio",
+        "lagomorfo",
+        "lupo",
+        "lince",
+        "marmotta",
+        "micromammifero",
+        "muflone",
+        "pecora",
+        "mustelide",
+        "uccello",
+        "orso",
+        "nutria",
+        "volpe",
+        "cinghiale",
+        "mucca",
+    ],
+    "de": [
+        "Dachs",
+        "Steinbock",
+        "Rothirsch",
+        "Gämse",
+        "Katze",
+        "Ziege",
+        "Rehwild",
+        "Hund",
+        "Eichhörnchen",
+        "Equiden",
+        "Ginsterkatze",
+        "Igel",
+        "Lagomorpha",
+        "Wolf",
+        "Luchs",
+        "Murmeltier",
+        "Kleinsäuger",
+        "Mufflon",
+        "Schaf",
+        "Mustelide",
+        "Vogen",
+        "Bär",
+        "Nutria",
+        "Fuchs",
+        "Wildschwein",
+        "Kuh",
+    ],
+}
+
+
+class OVMegaDetectorV5(pw_detection.MegaDetectorV5):
+    """MegaDetectorV5 class for detection model"""
+
+    def __init__(self, device):
+        self.model = OVModel("detection_model.xml", device)
+        self.device = "cpu"  # torch device, keep to CPU when using with OpenVINO
+
+    @staticmethod
+    def check_model():
+        """Check if model is initialized"""
+        return OVModel.check_model("detection_model.xml")
+
+    @staticmethod
+    def download_model(force: bool = False):
+        """Check if model is initialized"""
+        return OVModel.download_model("detection_model", force)
+
+
+class Classifier:
+    """Classifier class for classification model"""
+
+    CROP_SIZE = 182
+
+    def __init__(self, device):
```

Comment:
same here, do we want device default or we force providing it?

Reply:
As before